File: PracticeDijkstra.java

// https://www.programiz.com/dsa/dijkstra-algorithm
import java.util.Arrays;

public class PracticeDijkstra
{
    // Dijkstra's Algorithm in Java
    public static void dijkstra(int[][] graph, int source)
    {
        int count = graph.length;
        boolean[] visitedVertex = new boolean[count];
        int[] distance = new int[count];
        // int[] pi = new int[count];
        for (int i = 0; i < count; i++)
        {
            visitedVertex[i] = false;
            distance[i] = Integer.MAX_VALUE;
            //pi[i] = 0;
        }
        System.out.println(Arrays.toString(visitedVertex));
        System.out.println(Arrays.toString(distance));

        // Distance of self loop is zero
        distance[source] = 0;
        for (int i = 0; i < count; i++)
        {
            // Update the distance between neighbouring vertex and source vertex
            int u = findMinDistance(distance, visitedVertex);
            System.out.println("min: "+ u);
            visitedVertex[u] = true;

            // Update all the neighbouring vertex distances
            for (int v = 0; v < count; v++)
            {
                // node that has not been visited, and is connected to curr, and (currToStart + currToNext < nextToStart)
                if (!visitedVertex[v] && graph[u][v] != 0 && (distance[u] + graph[u][v] < distance[v]))
                {
                    System.out.println(v+": " +distance[u]+" "+graph[u][v]);
                    distance[v] = distance[u] + graph[u][v]; // update shorter path
                    //pi[v] = u;
                }
            }
            System.out.println(Arrays.toString(visitedVertex));
            System.out.println(Arrays.toString(distance));
            //System.out.println(Arrays.toString(pi));
            System.out.println();

        }
        for (int i = 0; i < distance.length; i++) {
            System.out.printf("Distance from %s to %s is %s%n", source, i, distance[i]);
        }

    }

    // Finding the minimum distance from graph row
    private static int findMinDistance(int[] distance, boolean[] visitedVertex)
    {
        int minDistance = Integer.MAX_VALUE;
        int minDistanceVertex = -1;
        for (int i = 0; i < distance.length; i++)
        {
            if (!visitedVertex[i] && distance[i] < minDistance)
            {
                minDistance = distance[i];
                minDistanceVertex = i;
            }
        }
        return minDistanceVertex;
    }

    public static void main(String[] args)
    {
        int graph[][] = new int[][]
                {
                { 0, 4, 2, 6, 8},
                { 0, 0, 0, 4, 3},
                { 0, 0, 0, 1, 0},
                { 0, 1, 0, 0, 3},
                { 0, 0, 0, 0, 0}};
        dijkstra(graph, 0);

    }

}

File: PracticeDijkstra.py
import sys


def PracticeDijkstra(g, source):
    count = len(g)
    visited = list(False for i in range(count))
    distance = list(sys.maxsize for i in range(count))
    distance[source] = 0
    for i in range(0, count):
        index = minDistance(distance, visited)
        visited[index] = True
        for j in range(0, count):
            if not visited[j] and g[index][j] != 0 and (distance[index]+g[index][j] < distance[j]):
                distance[j] = distance[index]+g[index][j]
    print(distance)


def minDistance(distance, visited):
    min = sys.maxsize
    minindex = -1
    for x in range(len(distance)):
        if not visited[x] and (distance[x] <=min):
            min = distance[x]
            minindex = x
    return minindex


g = [[0, 4, 2, 6, 8],
     [0, 0, 0, 4, 3],
     [0, 0, 0, 1, 0],
     [0, 1, 0, 0, 3],
     [0, 0, 0, 0, 0]]
PracticeDijkstra(g, 0)


